tests/test_protocols/test_generator/test_validate.py

```diff
@@ -378,33 +378,30 @@ def test_is_valid_dict(self):
 
     def test_is_valid_union(self):
         """Test for the '_is_valid_union' method."""
-        valid_content_type_1 = (
-            "pt:union[pt:bytes, pt:int, pt:float, pt:bool, pt:str, pt:set[pt:bytes], "
```

Comment:
Why changes here?

Reply:
@DavidMinarsch 
please check the generated result https://github.com/valory-xyz/open-aea/pull/572/files/0ddf5a1d3f5fdb4d621db60b71925f560efba75b#diff-f5a15260c42d9016ce3e20e557623e9f9e4d6cb2ae328e4f4a4bdbe621c9b2d8R109

as you see, every union value has own optional variable name and it's kind hard to handle deep nested data types.

implementation inst perfect, when i was working on, i just continued the initial idea found in the code.


current approach can be replaced with usage of `oneof` protobuf feature. but limits will stays the same